core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -527,13 +527,15 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    // New consumer always hasNext
-    override def hasData = true
-
-    override def receive() : BaseConsumerRecord = {
-      while (recordIter == null || !recordIter.hasNext)
+    override def hasData : Boolean = {
+      if (recordIter == null || !recordIter.hasNext) {
         recordIter = consumer.poll(1000).iterator
+      }
 
+      recordIter.hasNext
+    }
+
+    override def receive() : BaseConsumerRecord = {
```

Comment:
This fix works. There is another option. I noticed that the receive() method in NewShinnyConsumer class has changed. We can copy that implementation to let receive() throw ConsumerTimeoutException as well. This makes the implementation consistent.